onnxruntime/core/providers/cuda/object_detection/non_max_suppression_impl.cu

```diff
@@ -298,8 +300,9 @@ Status NmsGpu(std::function<IAllocatorUniquePtr<void>(size_t)> allocator,
       d_indices,           // input
       d_selected_boxes,    // selection flag
       d_selected_indices,  // selected items
-      d_num_selected, num_boxes));
-  CUDA_RETURN_IF_ERROR(cudaMemcpy(h_selected_count, d_num_selected, sizeof(int), cudaMemcpyDeviceToHost));
+      d_num_selected, num_boxes, stream));
+  CUDA_RETURN_IF_ERROR(cudaMemcpyAsync(h_selected_count, d_num_selected, sizeof(int), cudaMemcpyDeviceToHost, stream));
+  CUDA_RETURN_IF_ERROR(cudaStreamSynchronize(stream));
```

Comment:
>cudaStreamSynchronize [](start = 23, length = 21)

remove?

Reply:
> > cudaStreamSynchronize [](start = 23, length = 21)
> 
> remove? Please search for all cudaStreamSynchronize and check if it's needed. I think most of them are redundant.

cudaStreamSynchronize is needed since the value of h_selected_count will be used by host after this function.